sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -366,11 +367,11 @@ object AlluxioUtils extends Logging with Arm {
     val replaceFunc = genFuncForTaskTimeReplacement(pathsToReplace)
 
     files.map { file =>
-      val replacedFileInfo = replaceFunc(file.filePath)
+      val replacedFileInfo = replaceFunc(file.filePath.toString)
       if (replacedFileInfo.wasReplaced) {
         logDebug(s"TASK_TIME replaced ${file.filePath} with ${replacedFileInfo.fileStr}")
         PartitionedFileInfoOptAlluxio(PartitionedFile(file.partitionValues,
-          replacedFileInfo.fileStr, file.start, file.length),
+          GpuSparkPathShims.to(GpuSparkPath(replacedFileInfo.fileStr)), file.start, file.length),
```

Comment:
I find it a little odd to construct a GpuSparkPath only to pass it to a shim function that will turn it into something else.  Why doesn't the shim function take the string type directly?  It's cumbersome to type it this way, creates a throwaway object, and is a code pattern that shows up in a number of places.